### ISSUE TITLE
Question about "def Bilateral_NN"

### ISSUE THREAD

**Issue by JihyongOh** [ChenyangLEI/Fully-Automatic-Video-Colorization-with-Self-Regularization-and-Diversity#7]
Thank you for sharing your good code:)
I have a question about function named "Bilateral_NN".
1. What does the "division 500" mean for both color_image_X and color_image_X ??
2. Do the bold **5**s as below mean (relate) K=5??

def Bilateral_NN(color_image, neigh):
    h,w = color_image.shape[:2]    
    color_image_X = np.tile(np.arange(w),[h,1])/**500**.
    color_image_Y = np.tile(np.arange(h),[w,1]).T/**500**.
    color_image_all = np.concatenate([color_image,color_image_X[:,:,np.newaxis], color_image_Y[:,:,np.newaxis]],axis=2)
    neigh.fit(np.reshape(color_image_all, [-1, **5**]))
    idxs = neigh.kneighbors(np.reshape(color_image_all, [-1, **5**]), **5**, return_distance=False)
    return idxs

Thank you

**Comment by ChenyangLEI** [comment]
1. The weights for X, Y and RGB should be difference in bilateral space
2. Yes. Please check this code: https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html
